driver-core/src/main/java/com/datastax/driver/core/ExecutionInfo.java

```diff
@@ -87,7 +88,10 @@ ExecutionInfo withIncomingPayload(Map<String, ByteBuffer> incomingPayload) {
      * {@link com.datastax.driver.core.policies.RetryPolicy} may retry the
      * query on the same host, so the same host might appear twice.</li>
```

Comment:
Yes, consider the cost and compare it to the value for users. The question is, what percentage of users are interested in a detailed timeline of executions and retries, and how would they use it in their application?
For debugging => maybe trace logs are good enough
To compute metrics => we can expose those metrics ourselves, in a less invasive way (make them optional)